### PR TITLE
Set get pointer mode

### PR DESCRIPTION
**Pull request by amcamd**
- Add variable rocblas_pointer_mode to handle
- Add functions rocblas_set_pointer_mode and rocblas_get_pointer_mode
- Change enum name rocblas_mem_location to rocblas_pointer_mode

### CODE REVIEW

File: library/include/rocblas-auxiliary.h [hunk 1]
```diff
@@ -19,7 +19,7 @@
 /* ============================================================================================ */
 /*! \brief  indicates whether the pointer is on the host or device. currently HIP API can only recoginize the input ptr on deive or not
 can not recoginize it is on host or not */
-ROCBLAS_EXPORT rocblas_mem_location
+ROCBLAS_EXPORT rocblas_pointer_mode
 rocblas_get_pointer_location(void *ptr);
```
Comment (b-sumner) [inline library/include/rocblas-auxiliary.h]: This is not really the ideal name now.

Comment (amcamd) [inline library/include/rocblas-auxiliary.h]: An alternative would be to have two enums 1) rocblas_pointer_mode   2) rocblas_mem_location  . rocblas_pointer_mode would be set by the user to indicate if their pointers are on host or device. This is the same as the use in cuBLAS.  rocblas_mem_location would have its current usage, that is, the function would check where the pointer is, and use this to call the appropriate device kernel.

Comment (b-sumner) [inline library/include/rocblas-auxiliary.h]: Those wouldn't really be orthogonal.  I think one enum would be enough.  And the functions to access it should be intuitively named.

Comment (amcamd) [inline library/include/rocblas-auxiliary.h]: We have the functions
- rocblas_get_pointer_mode
- rocblas_set_pointer_mode
These set and get the variable rocblas_pointer_mode. 

This is an additional function that finds if a pointer is located on the device or the host. Would it be better to name it rocblas_pointer_location   ?

Comment (b-sumner) [inline library/include/rocblas-auxiliary.h]: Right, clearly get pointer mode is already taken.  But this function is returning a pointer mode.  How about rocblas_pointer_to_mode or...?

Comment (amcamd) [inline library/include/rocblas-auxiliary.h]: How about  rocblas_get_mode_of_pointer  ?

Then the functions would be
rocblas_get_pointer_mode, rocblas_set_pointer_mode, rocblas_get_mode_of_pointer

Comment (b-sumner) [inline library/include/rocblas-auxiliary.h]: I have a small preference for pointer_to_mode, but am not opposed to get_mode_of_pointer

Comment (amcamd) [inline library/include/rocblas-auxiliary.h]: new commit changes function name to    rocblas_pointer_to_mode

**Pull request merged**